src/GRA.Domain.Service/AvatarService.cs

```diff
@@ -431,6 +431,43 @@ public async Task<ICollection<AvatarElement>> GetUserAvatarAsync()
             return await _avatarElementRepository.GetUserAvatarAsync(GetActiveUserId());
         }
 
+        public async Task<Tuple<List<AvatarBundle>,List<string>>> GetUserUnlockBundles()
+        {
+            int activeUserId = GetActiveUserId();
+            var history = await _avatarBundleRepository.UserHistoryAsync(activeUserId);
+            var usersbundles = new List<AvatarBundle>();
+            var hasbeenviewed = new List<string>();
+            foreach (var item in history)
```

Comment:
`foreach`/`if` can be simplified: `foreach (var item in history.Where(_ => _.AvatarBundleId.HasValue && !_.IsDeleted)`

Reply:
I think everything in this `if` can be simplified to:
```
foreach(var bund in usersbundles.Where(_ => _.Id != item.AvatarBundleId))
{
    hasbeenviewed.Add(item.HasBeenViewed.ToString());
    usersbundles.Add(await GetBundleByIdAsync(item.AvatarBundleId.Value, false));
}
```

Reply:
It looks like this didn't make it in the last set of revisions.

Reply:
I modified this a little bit to work with the dictionary setup. But its pretty similar to what you have